lib/foreman_hooks/as_dependencies_hook.rb

```diff
@@ -2,6 +2,7 @@ module ForemanHooks
   module ASDependenciesHook
     def load_missing_constant(from_mod, constant_name)
       super(from_mod, constant_name).tap do |ret|
+        return unless ret.try(:name)
```

Comment:
We should keep the interface and return the original return value.
```suggestion
        return ret unless ret.try(:name)
```

Reply:
Also can the autoloaded constant be something that responds to `name`, but returns a falsey value?

Reply:
Unless `klass` can be a falsey value, we're good.
I don't think it's possible at all.